### PR TITLE
Keep enclosing braces of authors

### PR DESCRIPTION
**Pull request by koppor**
Fixes https://github.com/JabRef/jabref/issues/10031

Note that this PR also keeps braces inside the data model. A "better" data model would have a Boolean property with "protect", but we don't have. Thus, this is as good as it can get to fix the issue.

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Plaease check the MsBibExporter as well there is a manual workaround for such a case

**Comment by koppor** [comment]
I am working on it! Therefore no label "ready for review". Have to suspend work for now, will resume later.

File: src/main/java/org/jabref/model/entry/Author.java [hunk 1]
```diff
@@ -29,19 +29,29 @@ public class Author {
...
-     * @param jr        the junior part of the author's name (may consist of several tokens, like "Jr. III" in "Smith, Jr. III, John")
+     * @param firstPart     the first name of the author (may consist of several tokens, like "Charles Louis Xavier Joseph" in "Charles Louis Xavier Joseph de la Vall{\'e}e Poussin")
+     * @param firstAbbr the abbreviated first name of the author (may consist of several tokens, like "C. L. X. J." in "Charles Louis Xavier Joseph de la Vall{\'e}e Poussin"). It is a responsibility of the caller to create a reasonable abbreviation of the first name.
+     * @param vonPart       the von part of the author's name (may consist of several tokens, like "de la" in "Charles Louis Xavier Joseph de la Vall{\'e}e Poussin")
+     * @param lastPart      the last name of the author (may consist of several tokens, like "Vall{\'e}e Poussin" in "Charles Louis Xavier Joseph de la Vall{\'e}e Poussin")
+     * @param jrPart        the junior part of the author's name (may consist of several tokens, like "Jr. III" in "Smith, Jr. III, John")
```
Comment (calixtus) [inline src/main/java/org/jabref/model/entry/Author.java:39]: How about
`givenName` for `firstPart`
`prefix` for `vonPart`
`suffix` for `jrPart`
`familyName` for `lastPart`?
biblatex switched just a few years ago with the names. Was quite some effort to bugfix my styles.
See https://markov.htwsaar.de/tex-archive/macros/latex/contrib/biblatex/doc/biblatex.pdf#subsubsection.2.2.1

File: src/main/java/org/jabref/logic/importer/AuthorListParser.java [hunk 2]
```diff
@@ -94,9 +95,7 @@ private static StringBuilder buildWithAffix(Collection<Integer> indexArray, List
      * @param listOfNames the String containing the person names to be parsed
      * @return a parsed list of persons
      */
-    public AuthorList parse(String listOfNames) {
-        Objects.requireNonNull(listOfNames);
-
+    public AuthorList parse(@NonNull String listOfNames) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/importer/AuthorListParser.java:98]: As I told you before I am not a fan of this NonNull stuff, It servers the same purpose as the Objects.requiresNonNull

File: src/main/java/org/jabref/model/entry/Author.java (hunk 1 above)
Comment (koppor) [reply src/main/java/org/jabref/model/entry/Author.java:39]: ... When a small fix leads to a huge Refactoring...

Comment (Siedlerchr) [reply src/main/java/org/jabref/model/entry/Author.java:39]: Let's keep it for bibtex compat

Comment (koppor) [reply src/main/java/org/jabref/model/entry/Author.java:39]: Its only about variable names inside the code, not about parsing logic.

Comment (calixtus) [reply src/main/java/org/jabref/model/entry/Author.java:39]: In some cultures, the family name comes first. Bin latex changed the var names to avoid confusion of the developers as biblatex grew.

File: src/main/java/org/jabref/logic/importer/AuthorListParser.java (hunk 2 above)
Comment (koppor) [reply src/main/java/org/jabref/logic/importer/AuthorListParser.java:98]: Yeah, I know. We always have that debate :)

I'll collect pros and cons in an ADR.

TL;DR for me. It enables static code analysis for nulls - not some runtime exception stuff. JabRef should not die because of NPEs. 

Links - it was approved twice and did not cause any issues yet

- https://github.com/JabRef/jabref/pull/10840
- https://github.com/JabRef/jabref/pull/10521#issuecomment-1867142747

**Comment by koppor** [comment]
Working on MSBib makes it a follow-up to https://github.com/JabRef/jabref/pull/9729

File: src/main/java/org/jabref/model/entry/Author.java (hunk 1 above)
Comment (koppor) [reply src/main/java/org/jabref/model/entry/Author.java]: Renameings done in 92a20244e1

File: src/main/java/org/jabref/logic/msbib/MSBibMapping.java [hunk 3]
```diff
@@ -120,47 +128,36 @@ public class MSBibMapping {
         BIBLATEX_TO_MS_BIB.put(new UnknownField(MSBIB_PREFIX + "productioncompany"), "ProductionCompany");
     }
 
+    static {
+        BIB_ENTRYTYPE_MAPPING.put(StandardEntryType.Book, MSBibEntryType.Book);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/msbib/MSBibMapping.java:132]: Inconsistent usage, for the others you use Map.of...

Comment (koppor) [reply src/main/java/org/jabref/logic/msbib/MSBibMapping.java:132]: Map.of does not support that much parameters. I tried ,and. Checked jdk source

File: src/main/java/org/jabref/logic/exporter/MSBibExporter.java [hunk 4]
```diff
@@ -19,33 +19,37 @@
...
     @Override
-    public void export(final BibDatabaseContext databaseContext, final Path file,
-                       List<BibEntry> entries) throws SaveException {
-        Objects.requireNonNull(databaseContext);
-        Objects.requireNonNull(entries);
-
+    public void export(@NonNull BibDatabaseContext databaseContext,
+                       @NonNull Path file,
+                       @NonNull List<BibEntry> entries) throws SaveException {
+        Objects.requireNonNull(databaseContext); // required by test case
```
Comment (calixtus) [inline src/main/java/org/jabref/logic/exporter/MSBibExporter.java:40]: This is somekind of silly, isn't it?

File: src/main/java/org/jabref/logic/importer/fileformat/MsBibImporter.java [hunk 5]
```diff
@@ -24,24 +25,25 @@
 /**
  * Importer for the MS Office 2007 XML bibliography format
  */
+@NullMarked
 public class MsBibImporter extends Importer {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(MsBibImporter.class);
     private static final String DISABLEDTD = "http://apache.org/xml/features/disallow-doctype-decl";
     private static final String DISABLEEXTERNALDTD = "http://apache.org/xml/features/nonvalidating/load-external-dtd";
+    private static final DocumentBuilderFactory DOCUMENT_BUILDER_FACTORY = makeSafeDocBuilderFactory(DocumentBuilderFactory.newInstance());
```
Comment (calixtus) [inline src/main/java/org/jabref/logic/importer/fileformat/MsBibImporter.java:34]: The DocumentBuilderFactory as a constant? This smells funny, but also suspicious...

File: src/main/java/org/jabref/logic/openoffice/style/OOBibStyleGetCitationMarker.java [hunk 6]
```diff
@@ -20,8 +20,12 @@
 import org.jabref.model.openoffice.style.PageInfo;
 import org.jabref.model.strings.StringUtil;
 
+import org.jspecify.annotations.NonNull;
+
 class OOBibStyleGetCitationMarker {
 
+    private static final RemoveBracesFormatter REMOVE_BRACES_FORMATTER = new RemoveBracesFormatter();
```
Comment (calixtus) [inline src/main/java/org/jabref/logic/openoffice/style/OOBibStyleGetCitationMarker.java:27]: Again, looks weird, but I have no better idea...

File: src/main/java/org/jabref/logic/exporter/MSBibExporter.java (hunk 4 above)
Comment (koppor) [reply src/main/java/org/jabref/logic/exporter/MSBibExporter.java:40]: Future work would be to introduce JSPecify more globally - but I did not want to escalate the PR here.

File: src/main/java/org/jabref/logic/importer/fileformat/MsBibImporter.java (hunk 5 above)
Comment (koppor) [reply src/main/java/org/jabref/logic/importer/fileformat/MsBibImporter.java:34]: This has performance reasons. A Factory is instantiated once, but the produced things are instantiated more often (by the factory). - It takes time to setup the factory.

See https://community.sap.com/t5/technology-blogs-by-members/performance-improvements-in-nw-java-applications-with-xml-processing/ba-p/12894611 for more information.

> Solution - Make XML Factories Static

**Review by Siedlerchr — APPROVED** [review]
